src/java.base/share/classes/java/lang/StringUTF16.java

```diff
@@ -1532,27 +1528,23 @@ public static boolean contentEquals(byte[] value, CharSequence cs, int len) {
         return true;
     }
 
-    public static int putCharsAt(byte[] value, int i, char c1, char c2, char c3, char c4) {
+    public static void putCharsAt(byte[] value, int i, char c1, char c2, char c3, char c4) {
         int end = i + 4;
         checkBoundsBeginEnd(i, end, value);
-        putChar(value, i++, c1);
-        putChar(value, i++, c2);
-        putChar(value, i++, c3);
-        putChar(value, i++, c4);
-        assert(i == end);
-        return end;
+        putChar(value, i, c1);
+        putChar(value, i + 1, c2);
```

Comment:
Can someone please shed some light on this change: What is the benefit of `i + constant` over `i++`? 🤔

Reply:
i++ generates excessive store bytecodes compared to +1, +2, ... and the result bytecode size is larger, which is harmful to jit.